dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/JavaArchive.java

```diff
@@ -0,0 +1,575 @@
+	private static final int ZIP64_EOCDR_MINIMUM_SIZE = 56;
+	private static final int ZIP64_EOCDR_HEADER_SIZE = 12;           // 4 bytes for the signature and 8 bytes for the length
+	private static final int ZIP64_EOCDR_SIZE_OFFSET = 4;            // The size at this offset is the size of the zip64 EOCDR minus 12. (Meaning it does NOT include the 12-byte header)
+	private static final int ZIP64_EOCDR_SIZE_FIELD_LENGTH = 8;      // Number of bytes for size field.
+
+	// ZIP64_LOCATOR is Zip64-End-of-Central-Directory-Record-Locator
+	private static final int ZIP64_LOCATOR_SIGNATURE = 0x504b0607;
+	private static final int ZIP64_LOCATOR_SIZE = 20;                // Fixed size.
+	private static final int ZIP64_LOCATOR_EOCD_OFFSET_OFFSET = 8;   // Position in the locator of the offset of the Zip64-End-of-Central-Direcory-Record
+	
+	private static final int LOCAL_FILE_HEADER_SIGNATURE = 0x504b0304;
+	private static final int BLOCK_SIZE = 256;
+	private static final int FORWARD_BLOCK_SIZE = 8192;
+	
+	// Maximum amount of data (usually a script) that is allowed by this implementation to be 
+	// attached to the beginning of the archive. This value may be increased, but it will 
+	// increase the time it takes to fail when the zip is corrupted.
+	private static final int MAX_PREFIX_ALLOWED = 1024*1024*5;       
+	
+	private byte[] _endOfCentralDirectoryRecord; // Only the minimum size is stored.  Does not include the comment 
+	private int    _eocdRecordSize;              // Actual size of End-of-Central-Directory-Record - including comment
+	private byte[] _zip64EoCDRLocator;           // 20 bytes;
+	private long   _zip64EoCDRLocatorOffset;     // Offset (from end of file) of the Zip64 Locator
+	private byte[] _zip64EoCDR;                  // Only the minimum size is stored.  Does not include any extra fields.  
+	private long   _zip64EoCDRecSize;            // Size of Zip64-End-of-Central-Directory-Record
+    private String _archiveFileName;
+	private long   _fileLength;                  // Length of the archive file
+	
+    JavaArchive(String archiveFileName) {
+    	_archiveFileName = archiveFileName;
+    }
+    
+    public boolean isValid() {
+    	
+    	try (  RandomAccessFile file = new RandomAccessFile(_archiveFileName, "r") ) {
+
+    		_fileLength = file.length();
+    		_endOfCentralDirectoryRecord = getEndOfCentralDirectoryRecord(file);
+    		
+    		if ( isZip64() ) {
+    		   _zip64EoCDRLocator = getZip64EoCDLocator(file);
+    		   _zip64EoCDRLocatorOffset = _fileLength - _eocdRecordSize - ZIP64_LOCATOR_SIZE;
+    		   _zip64EoCDR = getZip64EndOfCentralDirectoryRecord(file);
+    		   _zip64EoCDRecSize = getZip64EoCDRecordSize( _zip64EoCDR, 0 );
+    		}
+    		
+    		int archiveOffset = (int)getActualArchiveOffset();
+
+    		if (archiveStartsWithSignature(file, archiveOffset)) {
+    			return true;
+    		} 
+
+    	} catch (Exception e) {
+    		//e.printStackTrace();
```

Comment:
Do we want to display some information here in case of an exception?

Reply:
Added // ffdc.   I think the ffdc is enough